### PR TITLE
clean spi/common dir

### PR DESCRIPTION
**Pull request by nlu90**
1. Move NetworkUtils and ShellUtils into utils/ dir

2. Move PackingPlan into packing/ dir

3. Fix import and dependency problems introduced by the above two changes

### CODE REVIEW

File: heron/spi/src/java/BUILD [hunk 1]
```diff
@@ -59,6 +59,7 @@ utils_deps_files = \
 scheduler_deps_files = \
     heron_java_proto_files() + [
         ":common-spi-java",
+        ":packing-spi-java",
```
Comment (maosongfu) [inline heron/spi/src/java/BUILD]: Move this to packing_deps_files below?

File: heron/statemgrs/src/java/BUILD [hunk 2]
```diff
@@ -6,6 +6,7 @@ common_deps_files = \
     heron_java_proto_files() + [
         "//heron/spi/src/java:common-spi-java",
         "//heron/spi/src/java:statemgr-spi-java",
+        "//heron/spi/src/java:utils-spi-java",
```
Comment (maosongfu) [inline heron/statemgrs/src/java/BUILD]: Let us not directly depend on utils-spi-java; it contains too many deep dependencies. 
Two options can be considered:
1. Add more specific building target in spi/src/java:BUILD
2. Re-org classes in utils, for instance, put SchedulerUtils into spi/src/java/scheduler or something similar.

File: heron/statemgrs/tests/java/BUILD [hunk 3]
```diff
@@ -12,6 +12,7 @@ spi_deps_files = [
     "//heron/common/src/java:basics-java", 
     "//heron/spi/src/java:common-spi-java", 
     "//heron/spi/src/java:statemgr-spi-java",
+    "//heron/spi/src/java:utils-spi-java",
```
Comment (maosongfu) [inline heron/statemgrs/tests/java/BUILD]: Let us not directly depend on utils-spi-java; it contains too many deep dependencies. 
Two options can be considered:
1. Add more specific building target in spi/src/java:BUILD
2. Re-org classes in utils, for instance, put SchedulerUtils into spi/src/java/scheduler or something similar.

File: heron/uploaders/src/java/BUILD [hunk 4]
```diff
@@ -4,6 +4,7 @@ uploader_spi_files = [
     "//heron/common/src/java:basics-java",
     "//heron/spi/src/java:common-spi-java",
     "//heron/spi/src/java:uploader-spi-java",
+    "//heron/spi/src/java:utils-spi-java",
```
Comment (maosongfu) [inline heron/uploaders/src/java/BUILD]: Let us not directly depend on utils-spi-java; it contains too many deep dependencies. 
Two options can be considered:
1. Add more specific building target in spi/src/java:BUILD
2. Re-org classes in utils, for instance, put SchedulerUtils into spi/src/java/scheduler or something similar.

File: heron/spi/src/java/BUILD (hunk 1 above)
Comment (nlu90) [inline heron/spi/src/java/BUILD]: The rule `scheduler-spi-java` depends on the PackingPlan.java class and the scheduler_deps_files is used to indicate the dependency for this rule.

**Comment by nlu90** [comment]
@maosongfu can you have a look?

**Comment by maosongfu** [comment]
looks good to me. 👍

**Pull request merged**